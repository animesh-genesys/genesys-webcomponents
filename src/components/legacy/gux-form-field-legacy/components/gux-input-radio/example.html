<form onchange="notify(event)" oninput="notify(event)">
  <h1>gux-form-field-legacy (Radio)</h1>

  <p>slot a input[type=radio] element into a gux-form-field-legacy element</p>

  <div>
    <gux-form-field-legacy>
      <input slot="input" type="radio" name="food" value="pizza" />
      <label slot="label">Pizza</label>
    </gux-form-field-legacy>
    <gux-form-field-legacy>
      <input slot="input" type="radio" name="food" value="pasta" checked />
      <label slot="label">Pasta</label>
    </gux-form-field-legacy>
    <gux-form-field-legacy>
      <input slot="input" type="radio" name="food" value="hamburger" />
      <label slot="label">Hamburger</label>
    </gux-form-field-legacy>
    <gux-form-field-legacy>
      <input slot="input" type="radio" name="food" value="sandwich" disabled />
      <label slot="label">Sandwich</label>
    </gux-form-field-legacy>
  </div>
</form>
