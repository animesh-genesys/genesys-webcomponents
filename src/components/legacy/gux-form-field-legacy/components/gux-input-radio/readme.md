# gux-input-radio

<!-- Auto Generated Below -->


## Slots

| Slot      | Description                           |
| --------- | ------------------------------------- |
| `"input"` | Required slot for input[type="radio"] |
| `"label"` | Required slot for label               |


## Dependencies

### Used by

 - [gux-form-field-legacy](../..)

### Graph
```mermaid
graph TD;
  gux-form-field-legacy --> gux-input-radio
  style gux-input-radio fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
