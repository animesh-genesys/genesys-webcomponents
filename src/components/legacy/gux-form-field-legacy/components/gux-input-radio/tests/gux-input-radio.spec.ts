import { newSpecPage } from '@stencil/core/testing';
import { GuxInputRadio } from '../gux-input-radio';

describe('gux-input-radio', () => {
  let page: SpecPage;

  beforeEach(async () => {
    page = await newSpecPage({
      components: [GuxInputRadio],
      html: `
      <gux-input-radio>
        <input slot="input" type="radio" name="dinner" value="sandwich" disabled>
        <label slot="label">Sandwich</label>
      </gux-input-radio>
      `,
      language: 'en'
    });
  });

  it('should build', async () => {
    expect(page.rootInstance).toBeInstanceOf(GuxInputRadio);
  });

  it('should render', async () => {
    expect(page.root).toMatchSnapshot();
  });
});
