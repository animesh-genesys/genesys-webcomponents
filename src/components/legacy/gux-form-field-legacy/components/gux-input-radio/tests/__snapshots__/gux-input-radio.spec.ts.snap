// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-input-radio should render 1`] = `
<gux-input-radio>
  <!---->
  <div class="gux-input-radio-container">
    <input disabled="" name="dinner" slot="input" type="radio" value="sandwich">
    <label slot="label">
      Sandwich
    </label>
  </div>
</gux-input-radio>
`;
